src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -17,487 +16,261 @@ import PlotStyling from '../../filter-cells/components/PlotStyling';
-                },
-                shape: { value: config.pointStyle },
-                fillOpacity: { value: config.pointOpa / 10 },
-              },
-            },
-          },
-          {
-            type: 'text',
-            from: { data: 'embeddingCat' },
-            encode: {
-              enter: {
-                x: { scale: 'x', field: 'um1' },
-                y: { scale: 'y', field: 'um2' },
-
-                text: {
-                  field: 'cluster_id',
-
-                },
-
-                fontSize: { value: config.labelSize },
-                strokeWidth: { value: 1.2 },
-                fill: { value: config.masterColour },
-                fillOpacity: { value: config.labelShow },
-                font: { value: config.masterFont },
-
-              },
-              transform: [
-                { type: 'label', size: ['width', 'height'] }],
-            },
-          },
-        ],
-
-        legends: config.legend,
-
-        title:
-        {
-          text: { value: config.titleText },
-          color: { value: config.masterColour },
-          anchor: { value: config.titleAnchor },
-          font: { value: config.masterFont },
-          dx: 10,
-          fontSize: { value: config.titleSize },
-        },
-      };
+import PlatformError from '../../../../../../components/PlatformError';
+import {
+  updatePlotConfig,
+  loadPlotConfig,
+} from '../../../../../../redux/actions/componentConfig';
+import { initialPlotConfigStates } from '../../../../../../redux/reducers/componentConfig/initialState';
+import generateEmbeddingCategoricalSpec from '../../../../../../utils/plotSpecs/generateEmbeddingCategoricalSpec';
+import generateEmbeddingContinuousSpec from '../../../../../../utils/plotSpecs/generateEmbeddingContinuousSpec';
+import colorProvider from '../../../../../../utils/colorProvider';
+import DimensionsRangeEditor from '../../../plots-and-tables/components/DimensionsRangeEditor';
+import ColourbarDesign from '../../../plots-and-tables/components/ColourbarDesign';
+import ColourInversion from '../../../plots-and-tables/components/ColourInversion';
+import LogExpression from '../../../plots-and-tables/embedding-continuous/components/LogExpression';
+import AxesDesign from '../../../plots-and-tables/components/AxesDesign';
+import PointDesign from '../../../plots-and-tables/components/PointDesign';
+import TitleDesign from '../../../plots-and-tables/components/TitleDesign';
+import FontDesign from '../../../plots-and-tables/components/FontDesign';
+import LegendEditor from '../../../plots-and-tables/components/LegendEditor';
+import LabelsDesign from '../../../plots-and-tables/components/LabelsDesign';
+
+const { Panel } = Collapse;
+
+// TODO: when we want to enable users to create their custom plots,
+// we will need to change this to proper Uuid
+// const plotUuid = 'embeddingCategoricalMain';
+// const plotType = 'embeddingCategorical';
+// const embeddingType = 'umap';
```

Comment:
is the plotUuid used now `embeddingCategorical` and `embeddingContinuous`?

Reply:
It isn't, that comment is not useful right now. We will have to do something with the UUIDs and the type of plots when we persist the configs to Redux, but for now it's not relevant to the code at all.